src/pages/List.js

```diff
@@ -17,27 +18,37 @@ export default function List({ token }) {
   }
 
   const markItemPurchased = (e, id, itemData) => {
-    const currentTimestamp = Date.now();
-    const initialInterval = itemData.purchase_frequency * 86400000;
-    const latestInterval = currentTimestamp - itemData.last_purchased;
+    // use DateTime package to get current time
+    const now = DateTime.now();
 
-    // const markItemPurchased = (e, id) => {
-    //   const elapsedMilliseconds = Date.now();
+    // convert now to readable ISO string
+    const nowToString = now.toString();
+
+    // initialize how many milliseconds there are in a day for calculation
+    const millisecondsInADay = 86400000;
+
+    // because last_estimate is an integer representing days, convert now to days
+    const nowInDays = Math.floor(now.ts / millisecondsInADay);
+
+    // do the same conversion for last_purchased
+    const lastPurchasedToDays = Math.floor(
+      DateTime.fromISO(itemData.last_purchased).ts / millisecondsInADay,
+    );
+
+    // determine the amount of days between now and when item was last marked purchased
+    const latestInterval = nowInDays - lastPurchasedToDays;
```

Comment:
This code's purpose is to provide `latestInterval`, the amount of days between now and when an item was last marked purchased.

Our operands, the things we need to determine our answer are 1) `itemData.last_purchased` and 2) some representation of the current time.

- Our data source `itemData.last_purchased` is in an ISO timestamp (e.g. `2021-05-21T22:34:59.167-07:00`).
- Our representation of the other input, the current timestamp can be whatever we want
- Our output needs to be days

Currently the solution is doing unnecessary conversion to their native formats to milliseconds, and converting them to days, including troubling ourselves with a magic number `86400000`.

We can accomplish the same with the following, utilizing the power of the Luxon library instead of doing gymnastics with converting into and out of their format:
```javascript
import { DateTime, Interval } from 'luxon';

export default function List({ token }) {
  const history = useHistory();
  const [listItems, loading, error] = useCollection(db.collection(token), {
    snapshotListenOptions: { includeMetadataChanges: true },
  });
  const [query, setQuery] = useState('');

  function handleReset() {
    setQuery('');
  }

  const markItemPurchased = (e, id, itemData) => {
    const currentDateTime = DateTime.now();

    const latestInterval = Math.floor(
      Interval.fromDateTimes(itemData.last_purchased, currentDateTime)
        .toDuration('days')
        .toObject().days,
    );

    if (e.target.checked === true) {
      // if an item has not yet been purchased, last_estimate has no value, so we initialize with the user's selected purchase_frequency
      if (itemData.times_purchased === 0) {
        db.collection(token)
          .doc(id)
          .update({
            last_purchased: currentDateTime.toString(),
            times_purchased: itemData.times_purchased + 1,
            last_estimate: itemData.purchase_frequency,
          });
        // if an item has at least 1 times_purchased, we use the last_estimate property to update the database last_estimate property
      } else {
        const latestEstimate = calculateEstimate(
          itemData.last_estimate,
          latestInterval,
          itemData.times_purchased,
        );
        db.collection(token)
          .doc(id)
          .update({
            last_purchased: currentDateTime.toString(),
            times_purchased: itemData.times_purchased + 1,
            last_estimate: latestEstimate,
          });
      }
    }
  };
```

Reply:
I got close to your suggested edits. I had to move `latestInterval` inside the `else` so that `itemData.last_purchased` isn't `null`.

And I had to wrap `itemData.last_purchased` in `DateTime.fromISO(itemData.last_purchased)`.

Reply:
Thank you SO MUCH @skylerwshaw for helping me with this particular functionality from Luxon. I KNEW it was possible to use, but I couldn't work out from the docs on my own HOW implement `Interval`. It makes so much sense now that I see it. But it was just out of my grasp before. I'm super grateful for your direction and help with this!

```javascript
 const calculateLatestInterval = (lastPurchased, currentDateTime) => {
    return Math.floor(
      Interval.fromDateTimes(DateTime.fromISO(lastPurchased), currentDateTime)
        .toDuration('days')
        .toObject().days,
    );
  };
```